packages/shopify-codemod/transforms/iife-to-ternary-expression.js

```diff
@@ -0,0 +1,48 @@
+export default function iifeToTernaryExpression({source}, {jscodeshift: j}, {printOptions = {}}) {
+  function isImmediatelyInvokedFunctionExpression(expressionPath) {
+    let expression = expressionPath.node.expression
+    return expression.type === 'CallExpression' && expression.callee.type === 'ArrowFunctionExpression'
+  }
+
+  function containsOnlyIfStatement(functionNode) {
+    return nodeIsBlockStatementContainingOnly(functionNode.body, 'IfStatement');
+  }
+
+  function onlyReturnsOneOfTwoValues(ifStatementNode) {
+    return nodeIsBlockStatementContainingOnly(ifStatementNode.consequent, 'ReturnStatement') &&
+           nodeIsBlockStatementContainingOnly(ifStatementNode.alternate, 'ReturnStatement');
+  }
+
+  function nodeIsBlockStatementContainingOnly(node, childType) {
+    return node.type === 'BlockStatement' &&
+           node.body.length === 1 &&
+         node.body[0].type === childType;
+  }
+
+  function isConvertibleExpression(path) {
+    return isImmediatelyInvokedFunctionExpression(path) &&
+           containsOnlyIfStatement(path.node.expression.callee) &&
+           onlyReturnsOneOfTwoValues(path.node.expression.callee.body.body[0]);
+  }
+
+  const sourceAST = j(source);
+
+  sourceAST
+    .find(j.ExpressionStatement)
+    .filter(isConvertibleExpression)
```

Comment:
Trying to avoid compacting things into a vague method name.  As you pointed out in my args spread PR, it's easier to read a few well-named, chained methods than it is to jump around from transform, to method, then back again.

Something like
```js
.filter(isIIFE)
.filter(isCalleeIfStatementWithReturns)
```